### PR TITLE
FEATURE: implement groups in SelectBox

### PR DESCRIPTION
**Pull request by klfman**

### CODE REVIEW

File: packages/react-ui-components/src/SelectBox/selectBox.js [hunk 1]
```diff
@@ -187,19 +191,57 @@ export default class SelectBox extends PureComponent {
                         }
                     </DropDown.Header>
                     <DropDown.Contents className={theme.dropDown__contents} scrollable={scrollable}>
-                        {(options || []).map(this.renderOption)}
+                        {Array.from(groupedOptions).map(this.renderGroup)}
```
[the remark this answers is not in the document]
Comment (klfman) [reply packages/react-ui-components/src/SelectBox/selectBox.js]: used Object.entries() as it made more sense

[the remark this answers is not in the document]
Comment (dimaip) [reply packages/react-ui-components/src/SelectBox/selectBox.js]: Do we polyfill `Object.entries` somewhere?

**Comment by dimaip** [comment]
Hey, any chances to fix the tests on this one?

**Comment by klfman** [comment]
Sorry, didn’t manage to do so today, will try on Monday

**Comment by skurfuerst** [comment]
I'll get them ready tomorrow

**Comment by skurfuerst** [comment]
superseded by #1081

**Pull request closed without merging**